src/main/java/org/fogbowcloud/saps/engine/core/model/SapsImage.java

```diff
@@ -33,16 +33,20 @@
 	private String user;
 	private int priority;
 	private String inputdownloadingTag;
+	private String digestInputdownloading;
 	private String preprocessingTag;
+	private String digestPreprocessing;
 	private String processingTag;
+	private String digestProcessing;
 	private Timestamp creationTime;
 	private Timestamp updateTime;
 	private String status;
 	private String error;
 
-	public SapsImage(String taskId, String dataset, String region, Date imageDate, ImageTaskState state, String arrebolJobId,
-					 String federationMember, int priority, String user, String inputdownloadingTag, String preprocessingTag,
-					 String processingTag, Timestamp creationTime, Timestamp updateTime, String status, String error) {
+	public SapsImage(String taskId, String dataset, String region, Date imageDate, ImageTaskState state,
```

Comment:
we need proper doc in this constructor (you can make in a refactor PR, latter)
do we need all these set methods here?
I don't link this SapsImage name. You should discuss a better option

Reply:
I created a problem (#75) to work on refactoring the code.